### ISSUE TITLE
When installing ASM on more then one cluster kubernetes_namespace tries to create the namespace only on cluster 

### ISSUE THREAD

**Issue by richiefrich** [terraform-google-modules/terraform-google-kubernetes-engine#1363]
### TL;DR

So this resource, unfortunately you can't specify the cluster name.  But when you have two clusters both needing ASM, this only makes the namespace on 1st cluster twice. 

```
resource "kubernetes_namespace" "system" {
  metadata {
    name = "istio-system"
  }
}
```

Here is the error I get.  

```
╷
│ Error: namespaces "istio-system" already exists
│
│   with module.asm-primary.kubernetes_namespace.system,
│   on .terraform/modules/asm-primary/modules/asm/main.tf line 34, in resource "kubernetes_namespace" "system":
│   34: resource "kubernetes_namespace" "system" {
│
╵
╷
│ Error: namespaces "istio-system" already exists
│
│   with module.asm-secondary[0].kubernetes_namespace.system,
│   on .terraform/modules/asm-secondary/modules/asm/main.tf line 34, in resource "kubernetes_namespace" "system":
│   34: resource "kubernetes_namespace" "system" {

```

Now if I go and remove both namespaces and run it again.  It will make the one namespace but only on the 1st clutser.

```
 # module.asm-primary.kubernetes_namespace.system will be created
  + resource "kubernetes_namespace" "system" {
      + id = (known after apply)

      + metadata {
          + generation       = (known after apply)
          + name             = "istio-system"
          + resource_version = (known after apply)
          + uid              = (known after apply)
        }
    }


  # module.asm-secondary[0].kubernetes_namespace.system will be created
  + resource "kubernetes_namespace" "system" {
      + id = (known after apply)

      + metadata {
          + generation       = (known after apply)
          + name             = "istio-system"
          + resource_version = (known after apply)
          + uid              = (known after apply)
        }
    }


module.asm-secondary[0].kubernetes_namespace.system: Creating...
module.asm-primary.kubernetes_namespace.system: Creating...


module.asm-secondary[0].module.cpr.module.gcloud_kubectl.null_resource.run_command[0] (local-exec): Error from server (NotFound): error when creating "STDIN": namespaces "istio-system" not found
```

So how can I fix this, thanks!

### Expected behavior

I expected that it said it created the namespace on cluster 2 but it fails not finding it. 

### Terraform Configuration

```hcl
// ASM
module "asm-primary" {
  source                    = "terraform-google-modules/kubernetes-engine/google//modules/asm"
  version                   = "21.1.0"
  project_id                = var.project_id
  cluster_name              = var.primary_cluster
  cluster_location          = var.primary_zone_1
  enable_cni                = true
}

module "asm-secondary" {
  count                     = lookup(local.secondary_enable[lookup(local.profile_type, var.tier)], "enable") ? 1 : 0

  source                    = "terraform-google-modules/kubernetes-engine/google//modules/asm"
  version                   = "21.1.0"
  project_id                = var.project_id
  cluster_name              = var.secondary_cluster
  cluster_location          = var.secondary_zone_1
  enable_cni                = true
}

```
```


### Terraform Version

```

$ terraform version
Terraform v1.1.9

```


### Additional information

_No response_

**Comment by richiefrich** [comment]
Any update on this please?

**Comment by bharathkkb** [comment]
Thanks for the report @richiefrich . To target different GKE in the same config, you will need to [alias](https://www.terraform.io/language/providers/configuration#alias-multiple-provider-configurations) the kubernetes provider and pass [into the module](https://www.terraform.io/language/providers/configuration#selecting-alternate-provider-configurations). Feel free to reopen if any issues.
```hcl
provider "kubernetes" {
alias = "asm-primary"
host = "https://${module.asm-primary.endpoint}"
...
}

provider "kubernetes" {
alias = "asm-secondary"
host = "https://${module.asm-secondary.endpoint}"
...
}

module "asm-primary" {
  source = "terraform-google-modules/kubernetes-engine/google//modules/asm"
...
  providers = {
    kubernetes = kubernetes.asm-primary
  }
}

module "asm-secondary" {
  source = "terraform-google-modules/kubernetes-engine/google//modules/asm"
...
  providers = {
    kubernetes = kubernetes.asm-secondary
  }
}

```
